src/Access/ExternalAuthenticators.cpp

```diff
@@ -270,23 +270,37 @@ void ExternalAuthenticators::setConfiguration(const Poco::Util::AbstractConfigur
 
     Poco::Util::AbstractConfiguration::Keys ldap_server_names;
     config.keys("ldap_servers", ldap_server_names);
-    for (const auto & ldap_server_name : ldap_server_names)
+    ldap_client_params_blueprint.clear();
+    for (auto ldap_server_name : ldap_server_names)
     {
         try
         {
-            ldap_client_params_blueprint.erase(ldap_server_name);
-            parseLDAPServer(ldap_client_params_blueprint.emplace(ldap_server_name, LDAPClient::Params{}).first->second, config, ldap_server_name);
+            const auto bracket_pos = ldap_server_name.find('[');
```

Comment:
How does `ldap_server_name` look like?  Why do you have to find that symbol and then resize the name...

Reply:
For multiple XML tags with the same name on the same level, Poco's config parser will represent them as keys with `[1]`, `[2]`, etc., appended to the tag name.

See this, for example:
https://github.com/ClickHouse/ClickHouse/blob/1e5aaa4c10747163b00b20c54a1d4642b43651b5/src/Access/AccessControl.cpp#L325

Reply:
I.e., for this:
```xml
<section>
    <subsection_A />
    <subsection_B />
    <subsection_A />
</section>
```
the keys would be names as: `subsection_A`, `subsection_B`, `subsection_A[2]`, or something like that.